rpcs3/Emu/SysCalls/Modules/cellMusic.cpp

```diff
@@ -99,9 +110,48 @@ s32 cellMusicSelectContents()
 	throw EXCEPTION("");
 }
 
-s32 cellMusicInitialize2()
+s32 cellMusicInitialize2(s32 mode, s32 spuPriority, vm::ptr<CellMusic2Callback> func, vm::ptr<void> userData)
 {
-	throw EXCEPTION("");
+	cellMusic.Todo("cellMusicInitialize2(mode=%d, spuPriority=%d, func=*0x%x, userData=*0x%x)", mode, spuPriority, func, userData);
+
```

Comment:
First one 100% (argument check), what else? I didn't notice "already initialized"-alike error code though. `fxm::make_always` will be fine for now. You create thread that exits immediately, this doesn't make much sense. Plus the cellMusic callback is "registered" (Register method), not "async". Async callbacks are mostly for the RSX thread, other usages are rough and to be replaced with more advanced implementations. Usually callbacks are just called from the PPU threads directly, if they are not "registered". Registered callbacks are called by cellSysutilCheckCallback, and this is usually explicitly mentioned. Asyncronous ones require PPU thread with fixed specified priority. cellAdec, cellVdec are rough examples, they are dirty though. When priority is implemented, it will be important how to call various types of callbacks.